hibernate-core/src/main/java/org/hibernate/type/BasicTypeRegistry.java

```diff
@@ -50,6 +50,11 @@ public BasicTypeRegistry(TypeConfiguration typeConfiguration){
 	}
 
 	public <J> BasicType<J> getRegisteredType(java.lang.reflect.Type javaType) {
+		if ( javaType instanceof Class ) {
+			// using `javaType.getTypeName()` causes problems with arrays
```

Comment:
What kind of problems? The type name of e.g. `int[]` should be `int[]` whereas the class name is `[I`. Maybe the types are registered under the wrong type name?

Reply:
I do not remember the specifics.  Maybe you have not seen the size of this change ;)

Regardless, the converse is true - making this change caused zero failures.  So I'm not sure why you changed it in the  first place

Reply:
It could also be related to primitive/wrapper "problems" which I also had to fix a bit